app/presenters/tree_node_builder.rb

```diff
@@ -62,108 +69,126 @@ def build
+    "ServiceResource"            => -> { generic_node(object.resource_name, object.resource_type == "VmOrTemplate" ? "vm.png" : "service_template.png") },
+    "ServerRole"                 => -> { server_role_node(object) },
+    "ServiceTemplate"            => -> { service_template_node },
+    "ServiceTemplateCatalog"     => -> { service_template_catalog_node },
+    "Snapshot"                   => -> { snapshot_node },
+    "Storage"                    => -> { generic_node(object.name, "storage.png") },
+    "Switch"                     => -> { generic_node(object.name,
                                                 "switch.png",
-                                                _("Switch: %{name}") % {:name => object.name})
-    when User                 then generic_node(object.name, "user.png")
-    when MiqSearch            then generic_node(object.description,
+                                                _("Switch: %{name}") % {:name => object.name}) },
+    "User"                       => -> { generic_node(object.name, "user.png") },
+    "MiqSearch"                  => -> { generic_node(object.description,
                                                 "filter.png",
-                                                _("Filter: %{filter_description}") % {:filter_description => object.description})
-    when MiqDialog            then generic_node(object.description, "miqdialog.png", object[0])
-    when MiqRegion            then miq_region_node
-    when MiqWidget            then generic_node(object.title, "#{object.content_type}_widget.png", object.title)
-    when MiqWidgetSet         then generic_node(object.name, "dashboard.png", object.name)
-    when Tenant               then generic_node(object.name,  "#{object.tenant? ? "tenant" : "project"}.png")
-    when VmdbTableEvm         then generic_node(object.name, "vmdbtableevm.png")
-    when VmdbIndex            then generic_node(object.name, "vmdbindex.png")
-    when Zone                 then zone_node
-    when Hash                 then hash_node
+                                                _("Filter: %{filter_description}") % {:filter_description => object.description}) },
+    "MiqDialog"                  => -> { generic_node(object.description, "miqdialog.png", object[0]) },
+    "MiqRegion"                  => -> { miq_region_node },
+    "MiqWidget"                  => -> { generic_node(object.title, "#{object.content_type}_widget.png", object.title) },
+    "MiqWidgetSet"               => -> { generic_node(object.name, "dashboard.png", object.name) },
+    "Tenant"                     => -> { generic_node(object.name, "#{object.tenant? ? "tenant" : "project"}.png") },
+    "VmdbTableEvm"               => -> { generic_node(object.name, "vmdbtableevm.png") },
+    "VmdbIndex"                  => -> { generic_node(object.name, "vmdbindex.png") },
+    "Zone"                       => -> { zone_node },
+    "Hash"                       => -> { hash_node },
+  }
```

Comment:
Nope, made it `build_node_hash` (local var) here since it was being manipulated in the next `.each`, and assign it as a constnat with `.freeze` it after that.  My way of getting around rubocop, but the `.sub` suggestion should solve that (hopefully), and allow us to assign as a constant from the get go.